core/client/src/main/java/alluxio/client/file/FileSystemMasterClient.java

```diff
@@ -185,7 +185,7 @@ public synchronized void delete(final AlluxioURI path, final DeleteOptions optio
     retryRPC(new RpcCallableThrowsAlluxioTException<Void>() {
       @Override
       public Void call() throws AlluxioTException, TException {
-        mClient.remove(path.getPath(), options.isRecursive());
+        mClient.remove(path.getPath(), options.toThrift());
```

Comment:
BTW, to make your proposed change backwards compatible, we should be adding an argument, not replacing an argument ... take a look at https://github.com/Alluxio/alluxio/pull/4645 for an example of how to do this